umd/index.d.ts

```diff
@@ -1,2 +1,458 @@
+     */
+    shiftLeft(numBits: number | Long): Long;
+  
+    /**
+     * Returns this Long with bits shifted to the left by the given amount.
+     */
+    shl(numBits: number | Long): Long;
+  
+    /**
+     * Returns this Long with bits arithmetically shifted to the right by the given amount.
+     */
+    shiftRight(numBits: number | Long): Long;
+  
+    /**
+     * Returns this Long with bits arithmetically shifted to the right by the given amount.
+     */
+    shr(numBits: number | Long): Long;
+  
+    /**
+     * Returns this Long with bits logically shifted to the right by the given amount.
+     */
+    shiftRightUnsigned(numBits: number | Long): Long;
+  
+    /**
+     * Returns this Long with bits logically shifted to the right by the given amount.
+     */
+    shru(numBits: number | Long): Long;
+  
+    /**
+     * Returns this Long with bits logically shifted to the right by the given amount.
+     */
+    shr_u(numBits: number | Long): Long;
+  
+    /**
+     * Returns this Long with bits rotated to the left by the given amount.
+     */
+    rotateLeft(numBits: number | Long): Long;
+  
+    /**
+     * Returns this Long with bits rotated to the left by the given amount.
+     */
+    rotl(numBits: number | Long): Long;
+  
+    /**
+     * Returns this Long with bits rotated to the right by the given amount.
+     */
+    rotateRight(numBits: number | Long): Long;
+  
+    /**
+     * Returns this Long with bits rotated to the right by the given amount.
+     */
+    rotr(numBits: number | Long): Long;
+  
+    /**
+     * Returns the difference of this and the specified Long.
+     */
+    subtract(subtrahend: number | Long | string): Long;
+  
+    /**
+     * Returns the difference of this and the specified Long.
+     */
+    sub(subtrahend: number | Long | string): Long;
+  
+    /**
+     * Converts the Long to a 32 bit integer, assuming it is a 32 bit integer.
+     */
+    toInt(): number;
+  
+    /**
+     * Converts the Long to a the nearest floating-point representation of this value (double, 53 bit mantissa).
+     */
+    toNumber(): number;
+  
+    /**
+     * Converts this Long to its byte representation.
+     */
+  
+    toBytes(le?: boolean): number[];
+  
+    /**
+     * Converts this Long to its little endian byte representation.
+     */
+  
+    toBytesLE(): number[];
+  
+    /**
+     * Converts this Long to its big endian byte representation.
+     */
+  
+    toBytesBE(): number[];
+  
+    /**
+     * Converts this Long to signed.
+     */
+    toSigned(): Long;
+  
+    /**
+     * Converts the Long to a string written in the specified radix.
+     */
+    toString(radix?: number): string;
+  
+    /**
+     * Converts this Long to unsigned.
+     */
+    toUnsigned(): Long;
+  
+    /**
+     * Returns the bitwise XOR of this Long and the given one.
+     */
+    xor(other: Long | number | string): Long;
+  }
+  
+  export default Long; // compatible with `import Long from "long"`
+  
```

Comment:
Maybe this can work instead?
```suggestion
export * from "long"
```

Reply:
No. This does not work for me.

```
error TS2709: Cannot use namespace 'Long' as a type.

39     i?: (number|Long|null);
                   ~~~~
```